products/appengine/api.yaml

```diff
@@ -63,3 +63,524 @@ objects:
           - UNSPECIFIED_ACTION
           - ALLOW
           - DENY
+  - !ruby/object:Api::Resource
+    name: 'StandardAppVersion'
+    description: |
+      A single firewall rule that is evaluated against incoming traffic
+      and provides an action to take on matched requests.
+    base_url: 'apps/{{project}}/versions'
```

Comment:
Thank you very much @rileykarson  I will be committing changes shortly.

Reply:
@rileykarson Need your help in something which I am struggling with.  I did try to use
the base_url path as apps/{{project}}/services/{{service}}/versions
but then I start getting 500 internal server error as the POST does not contain the name of the service e.g. "default"
/v1/apps/myapp/services//versions?alt=json

If I try to introduce a property called service then it fails with error: 
invalid or unknown key: service


```
resource "google_app_engine_standard_app_version" "myapp" {
  version_id = "v31"
  runtime = "nodejs10"
  api_config = {}
  endpoints_api_service = {}
  entrypoint {
    shell = "node index.js"
  }
```

your help is appreciated.

Reply:
Actually that was the original reason why I had to use the base_url as apps/{{project}}/versions
as this was the only way I could put the whole path inside the project parameter. But obviously that was wrong.
Conceptually I am struggling with the understanding of base_url and its connection with path parameters. How does the path parameters take the values. I see that "project" can come from provider , what about other parameters in the path? How they can be defined?
I did try adding the property service and setting 
```
exclude: true
```
in the terraform.yaml but doesn't seem to work either.

Reply:
I think I got it using 
```
- !ruby/object:Api::Type::ResourceRef
        name: 'service'
        resource: 'service'
        imports: 'name'
        description: |
          Service
        required: true  

```

Reply:
I am still struggling with this though moving forward.
So now I can reference the service from the Service resource. But I get the error:
"message": "Invalid JSON payload received. Unknown name \"service\" at 'version': Cannot find field.",

Looks like what I want to do is to exclude the "service" property from the Version but still have it in my parameters for the base_url.
I have been looking into examples from compute resource. 
Need some guidance on the yaml schema for the api.yml
and also understand the difference between, 
input: true
exclude: true
is_set: true

Your help is really appreciated.

Reply:
ignore above too, found url_param_only flag to not include in the body.
Missed it twice on reading. :)

Reply:
It looks like this part is still possibly unresolved:

> what about other parameters in the path? How they can be defined?

The `replaceVars` function we defined in the Terraform provider will pull the value from these sources, in order:

1. For special vars (`project`, `region`, `zone`) .tf config, then the provider config
2. .tf config (most fields are inferred here)
3. provider config (`config.go`)

Since `service` wasn't a property in Terraform previously, it couldn't be inferred from user config. Most APIs will silently ignore values that aren't really part of the request body, but this one doesn't. In those cases, yep, `url_param_only` is useful :+1: